docs/groovy/how-to-guides/built-in-variables.md

```diff
@@ -27,6 +30,66 @@ source = emptyTable(10).update(
 )
 ```
 
+## Alternatives for append-only tables
+
+When working with append-only tables where you need to reference preceding or following column values, avoid using `i` and `ii` with array notation. Instead, use one of the following approaches:
```

Comment:
add-only is different from append-only

add-only tables can have rows added in any position; append-only tables have rows only added at the end